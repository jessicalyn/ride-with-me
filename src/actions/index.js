export const hasError = (message) => ({
  type: 'HAS_ERROR',
  message
})

export const storeCities = (cities) => ({
  type: 'STORE_CITIES',
  cities
})
