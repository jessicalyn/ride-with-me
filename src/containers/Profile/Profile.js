import React, { Component } from 'react';
import { GoogleLogout } from 'react-google-login';
import { Redirect } from 'react-router-dom';

export class Profile extends Component {

  logout = () => {
    console.log("logout user")
    //erase user info in redux
  }

  render() {
    return(
      <div className="containers ride-search-container">
        { !this.props.user && <Redirect to='/login' />}
        <h2>My Profile</h2>
        <h4>Name</h4>
        <h4>Email</h4>
        <h4>Phone</h4>
        <h4>Password</h4>
        <h4>Add Image Upload?</h4>
        <GoogleLogout
          buttonText="Logout"
          onLogoutSuccess={this.logout}
        >
        </GoogleLogout>
      </div>
    )
  }
}
