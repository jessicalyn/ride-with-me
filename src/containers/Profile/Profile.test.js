import { Profile, mapStateToProps, mapDispatchToProps } from './Profile';
import React from 'react';
import { shallow } from 'enzyme';
import { logoutUser } from '../../actions';

jest.mock('../../actions')

describe('Profile Container', () => {
  describe('Profile', () => {
    let wrapper
    const mockUser = {id: 1, name: "Jessica"}
    let mockLogoutUser

    beforeEach(() => {
      mockLogoutUser = jest.fn()
      wrapper = shallow(
        <Profile 
          user={mockUser}
          logoutUser={mockLogoutUser}
        />
      )
    })

    it('should match the snapshot', () => {
      expect(wrapper.debug()).toMatchSnapshot()
    })

    it('should invoke logoutUser when logout is called', () => {
      wrapper.instance().logout()

      expect(mockLogoutUser).toHaveBeenCalled()
    })
  })

  describe('mapStateToProps', () => {
    it('should return the expected props to state', () => {

      const mockState = {
        user: {id: 1, name: "Jessica"},
        fakeState: "Not real state to return"
      }
      const expected = {
        user: {id: 1, name: "Jessica"}
      }
  
      const mappedProps = mapStateToProps(mockState)
      expect(mappedProps).toEqual(expected)
    })
  })

  describe('mapDispatchToProps', () => {
    it('should dispatch logoutUser', () => {
      const mockDispatch = jest.fn()
      const actionToDispatch = logoutUser()
      const mappedProps = mapDispatchToProps(mockDispatch)
  
      mappedProps.logoutUser()
  
      expect(mockDispatch).toHaveBeenCalled(actionToDispatch)
    })
  })
  
})