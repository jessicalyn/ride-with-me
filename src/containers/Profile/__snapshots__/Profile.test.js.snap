// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Profile Container Profile should match the snapshot 1`] = `
"<div className=\\"containers ride-search-container\\">
  <h2>
    My Profile
  </h2>
  <h4>
    Name: 
     
  </h4>
  <h4>
    Email: 
  </h4>
  <img src={[undefined]} alt=\\"User\\\\'s Google Profile\\" />
  <e clientId=\\"716984648143-q0uo1263pr96hegug4iqlotqslidqu9p.apps.googleusercontent.com\\" buttonText=\\"Logout\\" onLogoutSuccess={[Function]} type=\\"button\\" tag=\\"button\\" disabledStyle={{...}} icon={true} theme=\\"light\\" jsSrc=\\"https://apis.google.com/js/api.js\\" />
</div>"
`;
